core/capabilities/gateway_connector/service_wrapper.go

```diff
@@ -0,0 +1,156 @@
+package gatewayconnector
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"errors"
+	"math/big"
+	"slices"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/jonboulle/clockwork"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink/v2/core/config"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/api"
+	gwCommon "github.com/smartcontractkit/chainlink/v2/core/services/gateway/common"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/connector"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/network"
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore"
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore/keys/ethkey"
+)
+
+type ServiceWrapper struct {
+	services.StateMachine
+
+	config    config.GatewayConnector
+	keystore  keystore.Eth
+	connector connector.GatewayConnector
+	lggr      logger.Logger
+}
+
+type connectorSigner struct {
+	services.StateMachine
+
+	connector   connector.GatewayConnector
+	signerKey   *ecdsa.PrivateKey
+	nodeAddress string
+	lggr        logger.Logger
+}
+
+var _ connector.Signer = &connectorSigner{}
+
+func NewConnectorSigner(config config.GatewayConnector, signerKey *ecdsa.PrivateKey, lggr logger.Logger) (*connectorSigner, error) {
```

Comment:
Does it make sense for this to live in its own file?

Reply:
Bolek had asked for this func to be moved into here fwiw.

Reply:
I don't care too much. It's ultra-small, can be anywhere you guys prefer.

Reply:
I'm now realizing that Signer doesn't need Start()/Close() so you can remove those to further simplify.
Same for SetConnector().

Honestly maybe even go a step further and make ServiceWrapper just implement connector.Signer interface :)

Reply:
I also don't have a strong opinion here, reads slightly cleaner to me being separate, but I'll leave it up to your opinion David.

Reply:
> Honestly maybe even go a step further and make ServiceWrapper just implement connector.Signer interface :)

+1 for this, would be a lot cleaner